notebooks/_06-Architecture.md

```diff
@@ -0,0 +1,429 @@
+using matplotlib/seaborn. 
+* Tables will be generated as markdown tables, html tables, or any other way if found easier
+* Final presentation could be developed either on powerpoint or as a set of slides using jupyter
+
+## 16 Description of maintenance plan:
+### 16.1 How will the system / software be maintained after initial creation?
+
+The source code of the project will be improved/expanded by adding more capabilities as deemed needed during grad
+school
+
+### 16.2 What needs to be done for maintenance?
+The source code will be modified following the github workflow.
+
+### 16.3 Who will perform needed maintenance activities?
+Hassan/Amine, and other interested students in Joan's group.
+
+## 17. Description of project tools:
+### 17.1 What are all the tools and major external systems upon which the project depends?
+Standard scientific packages such as: numpy, pandas, scipy, pylogit, causalgraphicalmodels(or others),
+seaborn/matplotlib, scikitlearn, pyro, fitter, pytest. This list is not exhaustive and could grow.
+
+### 17.2 Is there evidence that we understand the basics of how all of those tools work?
+- **Have we read any/all relevant documentation and instruction materials?**
+- **Have we taken any basic training courses in any of these tools?**
+
+Each team members is individually familiar working with the majority of these packages. Collectively, the team is
+familiar working with all of these packages. If additional packages are needed and no team member has the expertise
+needed to work with them, the team will make sure to read any available documentation or use examples available. 
+
+## 18. Description of project risks and mitigation efforts:
+### 18.1  There should be some evidence for, and of, project feasibility:
+- **How do we know that every part of the architecture will work?**
+- **Based on the architecture, what are all needed resources?**
+
+The architecture is mainly based on third party dependencies that have proven to work. The needed resources will be
+one's laptop with an installed environments/requirements and compatible python version.
+
+### 18.2 What could render the project infeasible?
+- **Is there adequate resourcing?**
+
+We do not envision inadequare resourcing problems to occur.
+
+### 18.3 Are there known sources of difficulty based on the planned tasks?
+The sources of difficulty are related to learning proper software engineering practices and learning about new
+third party dependencies.
+
+### 18.4 Are there mismatches between prior knowledge and required tasks?
+Yes, this is mitigated by reading literature on needed methodology or documentation of any third party dependencies.
+
+### 18.5 What are the weakest areas of the architecture?
+- **Where is the project most vulnerable to failure?**
+- **How are these weaknesses being addressed?**
+At this moment, the weakest areas of the architecture are the lack of tests in the available code and the lack of 
```

Comment:
I would add two things.

1. The lack of a project workflow file to enable reproduction of the final results. I think this one will be fairly straightforward to make, assuming we follow good practices of creating modular notebooks and code.
2. The lack of a reproducible experiment system. We have no way to track all the different ideas that are likely about to be tried in the next month to get final results. Afterwards, we will then be unable/less-able to reproduce our work along the way to the final results. As a result, we'll be unable/less-able to harvest any useful information from this period since it's more difficult to use the knowledge of what didn't work if you don't remember all the things that didn't work.